lib/cassandra/tasks/seedregistry.rb

```diff
@@ -0,0 +1,161 @@
+require 'daemon_runner/shell_out'
+require 'daemon_runner/semaphore'
+
+module Cassandra
+  module Tasks
+   class SeedRegistry
+     # Create a new SeedRegistry task
+     #
+     # @param cluster_name [String] unique name (in Consul) for the Cassandra cluster
+     #
+     # @return [SeedRegistry]
+     #
+     def initialize cluster_name
+       @cluster_name = cluster_name.to_s
+       raise ArgumentError.new('cluster_name must not be empty') if @cluster_name.empty?
+       @semaphore = nil
+       @renew_thread = nil
+     end
+
+     # Schedule the sedd registration process to run periodically
+     #
+     def schedule
+       [:interval, '10m']
```

Comment:
This should be a much shorter interval, on the order of `10s`, and the session for the lock that this process acquires should have a short TTL (e.g. `20s`). This guards against:

* The Cassandra node being decommissioned during the 10 minute interval, leaving an invalid seed in place until the next check
* this utility crashing and leaving a possibly invalid lock sitting around indefinitely